pkg/controller/admissionchecks/multikueue/workload.go

```diff
@@ -436,20 +452,105 @@ func (w *wlReconciler) reconcileGroup(ctx context.Context, group *wlGroup) (reco
 		}
 	}
 
-	// finally - create missing workloads
+	return w.nominateAndSynchronizeWorkers(ctx, group, w.dispatcherName)
+}
+
+func (w *wlReconciler) nominateAndSynchronizeWorkers(ctx context.Context, group *wlGroup, dispatcherMode string) (reconcile.Result, error) {
+	log := ctrl.LoggerFrom(ctx).WithValues("op", "nominateAndSynchronizeWorkers")
+	log.V(3).Info("Nominate and Synchronize Worker Clusters")
+	var nominatedWorkers []string
+	var retryAfter time.Duration
+
+	switch dispatcherMode {
+	case config.MultiKueueDispatcherModeAllAtOnce:
+		for workerName := range group.remotes {
+			nominatedWorkers = append(nominatedWorkers, workerName)
+		}
+	case config.MultiKueueDispatcherModeIncremental:
+		key := types.NamespacedName{Name: group.local.Name, Namespace: group.local.Namespace}
+		roundStart, found := w.roundStartTimes[key]
+		now := w.clock.Now()
+
+		if found && now.Sub(roundStart) <= incrementalDispatcherRoundTimeout {
+			remainingWaitTime := incrementalDispatcherRoundTimeout - now.Sub(roundStart)
+			log.V(5).Info("Incremental Dispatcher nomination round still in progress", "remainingWaitTime", remainingWaitTime)
+			return reconcile.Result{RequeueAfter: remainingWaitTime}, nil
+		}
+
+		nextNominatedWorkers, err := getNextNominatedWorkers(log, group)
+		if err != nil {
+			log.V(2).Error(err, "nominating next worker clusters failed")
```

Comment:
```suggestion
			log.Error(err, "nominating next worker clusters failed")
```

Do we need level specification?

Reply:
Done